tests/kafkatest/services/kafka/kafka.py

```diff
@@ -778,6 +790,13 @@ def prop_file(self, node):
         for prop in self.per_node_server_prop_overrides.get(self.idx(node), []):
             override_configs[prop[0]] = prop[1]
 
+        if self.use_share_groups is not None and self.use_share_groups is True:
+            override_configs[config_property.SHARE_GROUP_ENABLE] = str(self.use_share_groups)
+            override_configs[config_property.UNSTABLE_API_VERSIONS_ENABLE] = str(self.use_share_groups)
+            override_configs[config_property.GROUP_COORDINATOR_REBALANCE_PROTOCOLS] = 'classic,consumer,share'
+            override_configs[config_property.SHARE_COORDINATOR_STATE_TOPIC_REPLICATION_FACTOR] = str(min(self.num_nodes, 3))
```

Comment:
I wonder why it is necessary to set the replication factor and min-isr for the share-state topic in this manner. I would have thought that the logic for setting values for the consumer-offsets and transaction-state topics would be the same as needed for the share-state topic. Please investigate more.

Personally, I think the `use_share_groups` setting should just turn on the configs needed to turn on share groups.